pyqt-apps/siriushla/li_di_bpms/bpm_main.py

```diff
@@ -0,0 +1,482 @@
+            graph_plot.addChannel(
+                y_channel=self.prefix + self.device_name + ':' + channel_data.get('path'),
+                name=channel_data.get('name'),
+                color=channel_data.get('color'),
+                lineWidth=1)
+
+        graph_plot.setMinimumWidth(600)
+        graph_plot.setMinimumHeight(250)
+
+        return graph_plot
+
+    def display_graph(self):
+        '''Display the graph tabs and all their contents'''
+
+        gp_vlay = QVBoxLayout()
+        tab = QTabWidget()
+        tab.setObjectName("LITab")
+
+        for graph_name in self.graph_all_data:
+            tablay = QVBoxLayout()
+            tab_content = QWidget()
+
+            graph_item = self.graph_all_data.get(graph_name)
+
+            if len(graph_item.items()) != 2:
+                tablay.addWidget(self.createGraph(graph_item), 10)
+                tab_content.setLayout(tablay)
+            else:
+                for data in graph_item:
+                    tablay.addWidget(
+                        self.createGraph(graph_item.get(data)), 10)
+
+                tab_content.setLayout(tablay)
+            tab.addTab(tab_content, graph_name)
+
+        gp_vlay.addWidget(tab)
+
+        return gp_vlay
+
+    def dataItem(self, channel, style):
+        '''Get data channel info'''
+
+        if style == 0:
+            channel_info = PyDMLabel(
+                parent=self,
+                init_channel=self.prefix + self.device_name + ':' + channel)
+        elif(style == 1 or style == 2 or style == 4):
+            channel_info = PyDMSpinbox(
+                parent=self,
+                init_channel=self.prefix + self.device_name + ':' + channel)
+            channel_info.showStepExponent = False
+        else:
+            channel_info = QLabel("Error", self)
+
+        return channel_info
+
+    def display_data(self, title, info, pos_x, pos_y, style):
+        '''Build a data widget'''
+
+        glay = QGridLayout()
+        group = QGroupBox()
+
+        countx = 0
+        county = 0
+
+        if style == 0:
+            for text, channel in info.items():
+
+                text_lb = QLabel(text, self)
+                glay.addWidget(text_lb, countx, county)
+
+                channel_lb = self.dataItem(channel, pos_y)
+                channel_lb.showUnits = True
+                glay.addWidget(channel_lb, countx, county+1)
+
+                countx += 1
+        else:
+            channel_lb = self.dataItem(info, pos_y)
+            channel_lb.showUnits = True
+            glay.addWidget(channel_lb, pos_x, pos_y)
+
+        glay.setAlignment(Qt.AlignCenter)
+
+        group.setTitle(title)
+        group.setLayout(glay)
+
+        return group
+
+    def display_mainData(self):
+        '''Display all main data widgets'''
+
+        countx = 0
+        county = 0
+
+        md_glay = QGridLayout()
+
+        for title, info in self.bpm_main_data.items():
+
+            if (title == "Trigger Cnt"
```

Comment:
do you really need this line break here? it seems the code would fit in a single line.